private/byte.rkt

```diff
@@ -109,3 +116,279 @@
+
+      (check-equal? (byte-ref (byte-or x (byte 0 0 0 0 0 1 0 0)) 5) 1)
+      (check-equal? (byte-ref (byte-or (byte 0 0 0 0 0 1 0 0) x) 5) 1)
+      
+      (check-equal? (byte-ref (byte-or x (byte 0 0 0 0 0 0 1 0)) 6) 1)
+      (check-equal? (byte-ref (byte-or (byte 0 0 0 0 0 0 1 0) x) 6) 1)
+
+      (check-equal? (byte-ref (byte-or x (byte 0 0 0 0 0 0 0 1)) 7) 1)
+      (check-equal? (byte-ref (byte-or (byte 0 0 0 0 0 0 0 1) x) 7) 1))))
+
+(define (byte-not b)
+  (apply byte
+         (for/list ([i (in-range 8)])
+           (bitwise-xor (byte-ref b i) 1)))) 
+
+(module+ test
+  (test-case
+    "byte-not"
+    (for ([x (in-range 256)])
+      (check-equal? (byte-ref (byte-not x) 0) (bitwise-xor (byte-ref x 0) 1))
+      (check-equal? (byte-ref (byte-not x) 1) (bitwise-xor (byte-ref x 1) 1))
+      (check-equal? (byte-ref (byte-not x) 2) (bitwise-xor (byte-ref x 2) 1))
+      (check-equal? (byte-ref (byte-not x) 3) (bitwise-xor (byte-ref x 3) 1))
+      (check-equal? (byte-ref (byte-not x) 4) (bitwise-xor (byte-ref x 4) 1))
+      (check-equal? (byte-ref (byte-not x) 5) (bitwise-xor (byte-ref x 5) 1))
+      (check-equal? (byte-ref (byte-not x) 6) (bitwise-xor (byte-ref x 6) 1))
+      (check-equal? (byte-ref (byte-not x) 7) (bitwise-xor (byte-ref x 7) 1)))))
+
+(define (byte-xor left right)
+  (apply byte
+         (for/list ([i (in-range 8)])
+           (bitwise-xor (byte-ref left i)
+                        (byte-ref right i))))) 
+(module+ test
+  (test-case
+    "byte-xor"
+    (for ([x (in-range 256)])
+      (check-equal? (byte-ref (byte-xor x 0) 0)
+                    (byte-ref x 0))
+      (check-equal? (byte-ref (byte-xor x 255) 0)
+                    (byte-ref (byte-not x) 0))
+
+      (check-equal? (byte-ref (byte-xor x 0) 1)
+                    (byte-ref x 1))
+      (check-equal? (byte-ref (byte-xor x 255) 1)
+                    (byte-ref (byte-not x) 1))
+
+      (check-equal? (byte-ref (byte-xor x 0) 2)
+                    (byte-ref x 2))
+      (check-equal? (byte-ref (byte-xor x 255) 2)
+                    (byte-ref (byte-not x) 2))
+
+      (check-equal? (byte-ref (byte-xor x 0) 3)
+                    (byte-ref x 3))
+      (check-equal? (byte-ref (byte-xor x 255) 3)
+                    (byte-ref (byte-not x) 3))
+
+      (check-equal? (byte-ref (byte-xor x 0) 4)
+                    (byte-ref x 4))
+      (check-equal? (byte-ref (byte-xor x 255) 4)
+                    (byte-ref (byte-not x) 4))
+
+      (check-equal? (byte-ref (byte-xor x 0) 5)
+                    (byte-ref x 5))
+      (check-equal? (byte-ref (byte-xor x 255) 5)
+                    (byte-ref (byte-not x) 5))
+
+      (check-equal? (byte-ref (byte-xor x 0) 6)
+                    (byte-ref x 6))
+      (check-equal? (byte-ref (byte-xor x 255) 6)
+                    (byte-ref (byte-not x) 6))
+
+      (check-equal? (byte-ref (byte-xor x 0) 7)
+                    (byte-ref x 7))
+      (check-equal? (byte-ref (byte-xor x 255) 7)
```

Comment:
I think many of the xor tests can be combined into a few simpler tests, like this:

```
(for ([x (in-range 256)])
  (check-equal? (byte-xor x 0) x)
  (check-equal? (byte-xor x 255) (byte-not x)))
```

That seems easier than doing eight separate checks (one for each bit) for both of these properties we want to test.